tests/test_datastructures.py

```diff
@@ -38,6 +39,28 @@ def test_distinct_groupers(self):
             ordered=False,
         )
 
+    def test_queryset_filter_for_distinct_groupers(self):
+        poll1_version = PollVersionFactory(
+            content__poll=self.initial_version.content.poll, state=PUBLISHED
+        )
+        poll2_version = PollVersionFactory()
+        PollVersionFactory(content__poll=poll2_version.content.poll)
+        PollVersionFactory(content__poll=poll2_version.content.poll)
```

Comment:
I'm not sure this data is adequate to really test this properly tbh. I mean, for example if you had a bug where the `distinct_groupers` method didn't have distinct groupers when filtering by additional params, I don't think this test would catch this.

Also, I'm wondering what happens in a situation like so...
Lets say you had these 4 objects in the db:
1 = poll1_grouper, archived
2 = poll1_grouper, published
3 = poll2_grouper, archived
4 = poll2_grouper, archived
And then you did `versionable.distinct_groupers(versions__state__in=ARCHIVED)`
What would happen? Obviously 2 would not be in the queryset, but what about 3 and 4? I guess only one of those would be in the queryset, but which one?

Reply:
Tbh, When there is a bug in `distinct_groupers` for distinct groupers that should be cover in outside this test. I added the filtering parameter which I tried to cover here.

Regarding the situation, you have mentioned. there should be two latest ARCHIVED objects in queryset.